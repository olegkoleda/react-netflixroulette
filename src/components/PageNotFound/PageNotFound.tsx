import { Box } from "atomic-layout";
import React from "react";
import { Link } from "react-router-dom";
import Heading from "../Heading";
import Logo from "../Logo";
import { StyledLogo, StyledError } from "./styled.pageNotFound";
import { StyledMain } from "../Main/styled.main";
import Button from "../Button";

const PageNotFound = () => {
  return (
    <>
      <Box
        as={StyledMain}
        flex
        justifyContent="center"
        alignItems="center"
        height={"calc(100vh - 4.5rem)"}
        flexDirection="column"
      >
        <StyledLogo>
          <Logo />
        </StyledLogo>
        <Heading>PAGE NOT FOUND</Heading>
        <StyledError>404</StyledError>
        <Link to="/">
          <Button secondary>Back to home page</Button>
        </Link>
      </Box>
    </>
  );
};

export default PageNotFound;
