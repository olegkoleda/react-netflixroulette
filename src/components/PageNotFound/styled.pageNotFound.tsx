import styled from "styled-components";

export const StyledLogo = styled.div`
  position: absolute;
  top: 1rem;
  left: 1rem;
`;

export const StyledError = styled.div`
  font-size: 10rem;
`;

