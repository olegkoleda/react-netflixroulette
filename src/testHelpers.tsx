import React from "react";
import renderer from "react-test-renderer";
import { theme } from "./styles/Theme";
import { ThemeProvider } from "styled-components";
import { BrowserRouter as Router } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { initialAppState } from "./store/reducers/rootReducer";

const middleWares = [thunk];
const mockStore = configureMockStore(middleWares);
const store = mockStore(initialAppState);

export function renderWithTheme(component: React.ReactNode) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export const reduxWrapper = (Component: typeof React.Component | React.FC, data = {}) => (
  <Provider store={store}>
    <Router>
      <Component {...data}/>
    </Router>
  </Provider>
);
