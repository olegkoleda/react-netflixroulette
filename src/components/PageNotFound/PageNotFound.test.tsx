import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import PageNotFound from "./PageNotFound";
import { renderWithTheme } from "../../testHelpers";

test("PageNotFound component renders correctly", () => {
  const component = renderWithTheme(
    <Router>
      <PageNotFound />
    </Router>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
