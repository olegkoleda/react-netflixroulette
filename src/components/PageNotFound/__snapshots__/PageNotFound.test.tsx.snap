// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound component renders correctly 1`] = `
<main
  className="styledmain__StyledMain-sc-1oh6sq-0 ihRYAc sc-AxjAm bkGhTV"
  height="calc(100vh - 4.5rem)"
>
  <div
    className="styledpageNotFound__StyledLogo-e7272s-0 fWcAhl"
  >
    <span>
      <span
        className="styledlogo__StyledLogo-sc-1suyno6-0 dKPWah"
        fontWeight={800}
      >
        netflix
      </span>
      <span
        className="styledlogo__StyledLogo-sc-1suyno6-0 eKILPN"
        fontWeight={400}
      >
        roulette
      </span>
    </span>
  </div>
  <h3
    className="Heading-sc-15b2y3d-0 fKqoJe"
  >
    PAGE NOT FOUND
  </h3>
  <div
    className="styledpageNotFound__StyledError-e7272s-1 iqNCsZ"
  >
    404
  </div>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="styledbutton__StyledButton-sc-7hz9n2-0 lnvFAg"
    >
      Back to home page
    </button>
  </a>
</main>
`;
